perun-core/src/main/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntry.java

```diff
@@ -163,6 +164,20 @@ public Facility getFacilityByName(PerunSession sess, String name) throws Interna
 		return facilities;
 	}
 
+	public List<Facility> getFacilitiesByAttribute(PerunSession sess, String attributeName, String attributeValue)
+			throws InternalErrorException, PrivilegeException, AttributeNotExistsException, WrongAttributeAssignmentException {
+		Utils.checkPerunSession(sess);
+
+		// Authorization
+		if(!AuthzResolver.isAuthorized(sess, Role.SERVICEUSER)) {
```

Comment:
I believe ``Role.SERVICEUSER`` would allow any service user to call this. Regarding rest of our API, you should check for FACILITYADMIN role and you should filter resulting list of facilities to return only those, where user is facility manager. Or to be more correct you should check, if user has READ right on attribute he search by on that facility. If not, do not return the facility.